### PR TITLE
Use reference style links in all markdown files

### PR DESCRIPTION
**Pull request by sabaimran**
### Issue
Addresses #101 

### Description
Updates all `.md` files in the repository to follow the [markdown format for reference-style links](https://www.markdownguide.org/basic-syntax/#reference-style-links). This work was started with the previous PR #122 and is closed with this PR. 

This update makes it easier to read the plain text markdown file for developers, as well as eases the process of making changes to the links if required. The label function also provides additional benefit in making the web page more accessible to users utilizing a screen reader.

Some of the links in the repo are for relative file paths, while some are web links. Some reference pages/documents, while some reference images. I feel that it's helpful to provide new line separation between these groups for ease of reading, otherwise the reference link section can get a bit cluttered in link-heavy files.

We chose to use language labels, rather than numeric labels, as this helps to disambiguate against footnotes. This also gives the link a handle that provides some context for the link without having to scroll down.

I also picked up a few small grammar/link fixes here and there as needed.

Please point out any missing files as well! I'm open to any style recommendations. 

### Testing
Compare with existing files and ensure that no functional changes have propagated. Most of the files should behave exactly the same way, except there would be hover text for some of the links based on the label specified in the reference link.

### CODE REVIEW

**Comment by sabaimran** [comment]
Sorry, I like small, incremental commits :sweat_smile: . I can squash my commits lol. Otherwise feel free to merge with squashing.

**Comment by keithrfung** [comment]
We squash merge every time. Keeps the commit history clean. This looks great. Will give more thorough review in a bit.

File: README.md [hunk 1]
```diff
@@ -75,6 +75,7 @@ A huge thank you to those who helped to contribute to this project so far, inclu
 - [Oxide Design][oxide-design]
 - Many teams within Microsoft
 
+<!-- Links -->
```
Comment (keithrfung) [inline README.md:78]: Thanks for keeping this consistent. Super helpful.

File: docs/basics/steps/1_Key_Ceremony.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ The Key Ceremony is broken into several high-level steps. Each Guardian must ann
 
 ### Attendance
 
-Guardians exchange all public keys and ensure each fellow guardian has received an election and auxiliary public key ensuring at all guardians are in attendance.
+Guardians exchange all public keys and ensure each fellow guardian has received an election and auxiliary public key ensuring that all guardians are in attendance.
```
Comment (keithrfung) [inline docs/basics/steps/1_Key_Ceremony.md:13]: Nice grammar correction

File: docs/spec/0.95.0/1_Overview.md [hunk 3]
```diff
@@ -1,4 +1,7 @@
-# ElectionGuard Specification ![Version 0.95.0](https://img.shields.io/badge/Version-v0.95.0-green)
+# ElectionGuard Specification ![Version 0.95.0][green-badge-0.95.0]
 
 !!! info "Download Specification"
     No web version of this specification is available at this time. [Download specification here](https://github.com/microsoft/electionguard/releases/tag/v0.95.0)
+
+<!-- Links -->
+[green-badge-0.95.0]: https://img.shields.io/badge/Version-v0.95.0-green
```
Comment (keithrfung) [inline docs/spec/0.95.0/1_Overview.md:7]: I think the badges are overkill, but the specification one is missed here.

**Review by keithrfung — CHANGES_REQUESTED** [review]

File: docs/spec/0.95.0/1_Overview.md (hunk 3 above)
Comment (sabaimran) [reply docs/spec/0.95.0/1_Overview.md:7]: Yeah badges are probably overkill, but I felt it was good for consistency. Do you think I should remove it? Will update the spec link.

**Review by keithrfung — APPROVED** [review]

**Pull request merged**